base/cvd/cuttlefish/host/commands/cvd/instances/instance_record.cpp

```diff
@@ -83,4 +96,94 @@ Result<Json::Value> LocalInstance::FetchStatus(std::chrono::seconds timeout) {
   return CF_EXPECT(FetchInstanceStatus(*this, timeout));
 }
 
+Result<void> LocalInstance::PressPowerBtn() {
+  Command cmd(
+      CF_EXPECT(HostToolTarget(host_artifacts_path()).GetPowerBtnBinPath()));
+
+  cmd.AddParameter("--instance_num=", id());
+  cmd.SetEnvironment({});
+  AddEnvironment(cmd, *this);
+
+  LOG(DEBUG) << "Executing: " << cmd.ToString();
+
+  siginfo_t infop;
+  cmd.Start().Wait(&infop, WEXITED);
+  CF_EXPECT(CheckProcessExitedNormally(infop));
+
+  return {};
+}
+
+Result<void> LocalInstance::Restart(std::chrono::seconds launcher_timeout,
+                                    std::chrono::seconds boot_timeout) {
+  SharedFD monitor_socket = CF_EXPECT(GetLauncherMonitor(launcher_timeout));
+
+  LOG(INFO) << "Requesting restart";
+  CF_EXPECT(RunLauncherAction(monitor_socket, LauncherAction::kRestart,
+                              launcher_timeout.count()));
+
+  LOG(INFO) << "Waiting for device to boot up again";
+  CF_EXPECT(WaitForRead(monitor_socket, boot_timeout.count()));
+  RunnerExitCodes boot_exit_code = CF_EXPECT(ReadExitCode(monitor_socket));
+  CF_EXPECT(boot_exit_code != RunnerExitCodes::kVirtualDeviceBootFailed,
+            "Boot failed");
+  CF_EXPECT(boot_exit_code == RunnerExitCodes::kSuccess,
+            "Unknown response" << static_cast<int>(boot_exit_code));
+
+  LOG(INFO) << "Restart successful";
+  return {};
+}
+
+Result<void> LocalInstance::PowerWash(std::chrono::seconds launcher_timeout,
+                                      std::chrono::seconds boot_timeout) {
+  SharedFD monitor_socket = CF_EXPECT(GetLauncherMonitor(launcher_timeout));
+
+  LOG(INFO) << "Requesting powerwash";
+  CF_EXPECT(RunLauncherAction(monitor_socket, LauncherAction::kPowerwash,
+                              launcher_timeout.count()));
+
+  LOG(INFO) << "Waiting for device to boot up again";
+  CF_EXPECT(WaitForRead(monitor_socket, boot_timeout.count()));
+  RunnerExitCodes boot_exit_code = CF_EXPECT(ReadExitCode(monitor_socket));
+  CF_EXPECT(boot_exit_code != RunnerExitCodes::kVirtualDeviceBootFailed,
+            "Boot failed");
+  CF_EXPECT(boot_exit_code == RunnerExitCodes::kSuccess,
+            "Unknown response" << static_cast<int>(boot_exit_code));
+
+  LOG(INFO) << "Powerwash successful";
+
+  return {};
+}
+
+Result<Json::Value> LocalInstance::ReadJsonConfig() const {
+  Json::CharReaderBuilder builder;
+  std::string config_file = instance_dir() + "/cuttlefish_config.json";
+  std::ifstream ifs(config_file);
+  std::string errorMessage;
+  Json::Value config;
+  bool parsed = Json::parseFromStream(builder, ifs, &config, &errorMessage);
+  CF_EXPECTF(std::move(parsed), "Could not read config file {}: {}",
+             config_file, errorMessage);
+  return config;
+}
```

Comment:
`cuttlefish_config.json` also becomes part of the public API, with the caveats mentioned above.